### ISSUE TITLE
ORA-00904: "v0"."GET_WKT": invalid identifier

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#14232]
Depending on how we compose functions, some Oracle calls of `Get_WKT()` and similar spatial functions on `SDO_GEOMETRY` types don't work. E.g. this query:

```sql
select (
  select xmlelement(
    name result,
    xmlagg(xmlelement(
      name record,
      xmlelement(
        name "v0",
        xmlattributes(
          case
            when "v0".Get_WKT() is null then 'true'
          end as "xsi:nil"
        ),
        "v0".Get_WKT()
      )
    ))
  )
  from (
    select null "v0"
    from DUAL
    union all
    select sdo_geometry(wkt => 'point(1 1)', srid => null) "a"
    from DUAL
  ) "t"
)
from DUAL
```

Is generated from a recently added `MULTISET` / `GEOMETRY` feature combination integration test (see https://github.com/jOOQ/jOOQ/issues/14195).

And it produces this error:

```
org.jooq.exception.DataAccessException: SQL [select (select xmlelement(name result, xmlagg(xmlelement(name record, xmlelement(name "v0", xmlattributes(case when "v0".Get_WKT() is null then 'true' end as "xsi:nil"), "v0".Get_WKT())))) from (select ? "v0" from DUAL union all select sdo_geometry(wkt => ?, srid => null) "a" from DUAL) "t") from DUAL -- SQL rendered with a free trial version of jOOQ 3.18.0-SNAPSHOT]; ORA-00904: "v0"."GET_WKT": ungültige ID

	at org.jooq_3.18.0-SNAPSHOT.ORACLE21C.debug(Unknown Source)
	at org.jooq.impl.Tools.translate(Tools.java:3369)
	at org.jooq.impl.DefaultExecuteContext.sqlException(DefaultExecuteContext.java:746)
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:357)
	at org.jooq.impl.AbstractResultQuery.fetchLazy(AbstractResultQuery.java:301)
	at org.jooq.impl.AbstractResultQuery.fetchLazyNonAutoClosing(AbstractResultQuery.java:322)
	at org.jooq.impl.SelectImpl.fetchLazyNonAutoClosing(SelectImpl.java:2861)
	at org.jooq.impl.ResultQueryTrait.fetchOne(ResultQueryTrait.java:509)
	at org.jooq.impl.Tools.attach(Tools.java:1582)
	at org.jooq.impl.DefaultDSLContext.fetchOne(DefaultDSLContext.java:4672)
	at org.jooq.impl.DefaultDSLContext.lambda$55(DefaultDSLContext.java:4692)
	at org.jooq.impl.Tools.attach(Tools.java:1582)
	at org.jooq.impl.DefaultDSLContext.fetchValue(DefaultDSLContext.java:4692)
	at org.jooq.test.all.testcases.MultisetTests.lambda$211(MultisetTests.java:1531)
	at org.jooq.test.jOOQAbstractTest.withDSLContext(jOOQAbstractTest.java:2154)
	at org.jooq.test.jOOQAbstractTest.bindAndInlineWithDSLContexts(jOOQAbstractTest.java:2147)
	at org.jooq.test.BaseTest.bindAndInlineWithDSLContexts(BaseTest.java:1810)
	at org.jooq.test.all.testcases.MultisetTests.lambda$210(MultisetTests.java:1524)
	at org.jooq.test.all.testcases.MultisetTests.lambda$467(MultisetTests.java:3057)
	at org.jooq.test.all.testcases.MultisetTests.lambda$470(MultisetTests.java:3075)
	at org.jooq.test.all.testcases.MultisetTests.testRecordLevelListenerResults(MultisetTests.java:3114)
	at org.jooq.test.all.testcases.MultisetTests.testRecordLevelListenerResults(MultisetTests.java:3102)
	at org.jooq.test.all.testcases.MultisetTests.testMultisetEmulations(MultisetTests.java:3075)
	at org.jooq.test.all.testcases.MultisetTests.testMultisetEmulations(MultisetTests.java:3057)
	at org.jooq.test.all.testcases.MultisetTests.assertMultisetDataType(MultisetTests.java:1523)
	at org.jooq.test.all.testcases.MultisetTests.testMultisetProjectionDataTypeGeometry(MultisetTests.java:1731)
	at org.jooq.test.jOOQAbstractTest.testMultisetProjectionDataTypeGeometry(jOOQAbstractTest.java:5553)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:93)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:40)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:529)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:756)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:452)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:210)
Caused by: java.sql.SQLSyntaxErrorException: ORA-00904: "v0"."GET_WKT": ungültige ID

	at oracle.jdbc.driver.T4CTTIoer11.processError(T4CTTIoer11.java:629)
	at oracle.jdbc.driver.T4CTTIoer11.processError(T4CTTIoer11.java:563)
	at oracle.jdbc.driver.T4C8Oall.processError(T4C8Oall.java:1230)
	at oracle.jdbc.driver.T4CTTIfun.receive(T4CTTIfun.java:771)
	at oracle.jdbc.driver.T4CTTIfun.doRPC(T4CTTIfun.java:298)
	at oracle.jdbc.driver.T4C8Oall.doOALL(T4C8Oall.java:511)
	at oracle.jdbc.driver.T4CPreparedStatement.doOall8(T4CPreparedStatement.java:162)
	at oracle.jdbc.driver.T4CPreparedStatement.executeForDescribe(T4CPreparedStatement.java:1009)
	at oracle.jdbc.driver.OracleStatement.prepareDefineBufferAndExecute(OracleStatement.java:1270)
	at oracle.jdbc.driver.OracleStatement.executeMaybeDescribe(OracleStatement.java:1148)
	at oracle.jdbc.driver.OracleStatement.executeSQLSelect(OracleStatement.java:1660)
	at oracle.jdbc.driver.OracleStatement.doExecuteWithTimeout(OracleStatement.java:1469)
	at oracle.jdbc.driver.OraclePreparedStatement.executeInternal(OraclePreparedStatement.java:3760)
	at oracle.jdbc.driver.OraclePreparedStatement.execute(OraclePreparedStatement.java:4135)
	at oracle.jdbc.driver.OraclePreparedStatementWrapper.execute(OraclePreparedStatementWrapper.java:1013)
	at org.jooq.tools.jdbc.DefaultPreparedStatement.execute(DefaultPreparedStatement.java:219)
	at org.jooq.tools.jdbc.DefaultPreparedStatement.execute(DefaultPreparedStatement.java:219)
	at org.jooq.impl.Tools.executeStatementAndGetFirstResultSet(Tools.java:4588)
	at org.jooq.impl.AbstractResultQuery.execute(AbstractResultQuery.java:236)
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:343)
	... 55 more
Caused by: Error : 904, Position : 170, Sql = select (select xmlelement(name result, xmlagg(xmlelement(name record, xmlelement(name "v0", xmlattributes(case when "v0".Get_WKT() is null then 'true' end as "xsi:nil"), "v0".Get_WKT())))) from (select :1  "v0" from DUAL union all select sdo_geometry(wkt => :2 , srid => null) "a" from DUAL) "t") from DUAL -- SQL rendered with a free trial version of jOOQ 3.18.0-SNAPSHOT, OriginalSql = select (select xmlelement(name result, xmlagg(xmlelement(name record, xmlelement(name "v0", xmlattributes(case when "v0".Get_WKT() is null then 'true' end as "xsi:nil"), "v0".Get_WKT())))) from (select ? "v0" from DUAL union all select sdo_geometry(wkt => ?, srid => null) "a" from DUAL) "t") from DUAL -- SQL rendered with a free trial version of jOOQ 3.18.0-SNAPSHOT, Error Msg = ORA-00904: "v0"."GET_WKT": ungültige ID

	at oracle.jdbc.driver.T4CTTIoer11.processError(T4CTTIoer11.java:636)
	... 74 more
```

The solution is to wrap `"v0"` as `("v0").Get_WKT()`

**Comment by lukaseder** [comment]
This works:

```sql
select (
  select xmlelement(
    name result,
    xmlagg(xmlelement(
      name record,
      xmlelement(
        name "v0",
        xmlattributes(
          case
            when "v0".Get_WKT() is null then 'true'
          end as "xsi:nil"
        ),
        "v0".Get_WKT()
      )
    ))
  )
  from (
    select null "v0"
    from DUAL
    union all
    select sdo_geometry(wkt => 'point(1 1)', srid => null) "a"
    from DUAL
  ) "t"
)
from DUAL
```

And produces the expected output:

```xml
<RESULT>
	<RECORD>
		<v0 xsi:nil="true"></v0>
	</RECORD>
	<RECORD>
		<v0>POINT (1.0 1.0)</v0>
	</RECORD>
</RESULT>
```

**Comment by lukaseder** [comment]
The following functions were updated:

- `StAsBinary`
- `StAsText`
- `StGeometryType`
- `StIntersects`
- `StSrid`
- `StX`
- `StY`
- `StZ`

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.18.0 and 3.17.6 (https://github.com/jOOQ/jOOQ/issues/14233). A backport to 3.16 was omitted, as the related changes from https://github.com/jOOQ/jOOQ/issues/14195 are too significant and too much has been changed in the area of `MULTISET`
